typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,29 @@
+const $TabContainer = document.querySelector('.tab-container');
+const $AllTab = document.querySelectorAll('.tab');
+const $AllView = document.querySelectorAll('.view');
+
+if (!$TabContainer) throw new Error("Didn't work:(");
+
+function handleClick(event: Event): void {
+  const $eventTarget = event.target as HTMLDivElement;
+
+  for (let i = 0; i < $AllTab.length; i++) {
+    if ($eventTarget.textContent === $AllTab[i].textContent) {
+      $AllTab[i].className = 'tab active';
+    } else {
+      $AllTab[i].className = 'tab';
+    }
+  }
+
+  const dataView = $eventTarget.getAttribute('data-view');
+
+  for (let i = 0; i < $AllTab.length; i++) {
+    if (dataView !== $AllView[i].getAttribute('data-view')) {
+      $AllView[i].className = 'view hidden';
+    } else {
+      $AllView[i].className = 'view';
+    }
+  }
+}
+
+if ($TabContainer) $TabContainer.addEventListener('click', handleClick);
```

Comment:
Do you need this `if` statement here? You already have your `if` statement that essentially checks that `$TabContainer` is not `null` so this shouldn't be necessary.